frontend/javascripts/admin/time/time_line_view.js

```diff
@@ -109,9 +131,57 @@ class TimeLineView extends React.PureComponent<*, State> {
     this.fetchTimeTrackingData();
   };
 
+  getTooltipForEntry(taskId: string, start: Date, end: Date) {
+    const isSameDay = start.getUTCDate() === end.getUTCDate();
+    const duration = end - start;
+    const durationAsString = formatDurationToHoursAndMinutes(duration);
+    const dayFormatForMomentJs = "MMM, DD, YYYY";
+    const tooltip = (
+      <div className="google-charts-tooltip">
+        <div className="highlighted">
+          Task ID: {taskId}
+          <div className="striped-border" />
+        </div>
+
+        <table>
+          <tbody>
+            <tr>
+              <td className="highlighted">Date:</td>
+              <td>
+                {isSameDay ? (
+                  <FormattedDate timestamp={start} format={dayFormatForMomentJs} />
+                ) : (
+                  <React.Fragment>
+                    <FormattedDate timestamp={start} format={dayFormatForMomentJs} /> -{" "}
+                    <FormattedDate timestamp={end} format={dayFormatForMomentJs} />
+                  </React.Fragment>
+                )}
+              </td>
+            </tr>
+            <tr>
+              <td className="highlighted">Time:</td>
+              <td>
+                <FormattedDate timestamp={start} format={hourFormat} /> -{" "}
+                <FormattedDate timestamp={end} format={hourFormat} />
+              </td>
+            </tr>
+            <tr>
+              <td className="highlighted">Duration:</td>
+              <td>{durationAsString}</td>
+            </tr>
+          </tbody>
+        </table>
+      </div>
+    );
+    return ReactDOMServer.renderToStaticMarkup(tooltip);
+  }
+
   render() {
     const columns = [
       { id: "AnnotationId", type: "string" },
+      // This label columns is somehow needed to make the custom tooltip work.
+      { type: "string", id: "empty label" },
```

Comment:
Interesting... Out of curiosity, how did you find that workaround? Was it posted somewhere? If so, you could add the link so that we can follow-up with this later (maybe this will be fixed in another version of the plotting lib).

Reply:
The actual hint was here [https://developers.google.com/chart/interactive/docs/gallery/timeline#customizing-tooltips](https://developers.google.com/chart/interactive/docs/gallery/timeline#customizing-tooltips):

> Hovering over a bar brings up a tooltip with the text defined in the third column. In this chart, we have to set the second column to dummy values (here, null) so that our tooltips can exist in the third column.

Imo This does not make any sense, but it finally worked after reading this :slightly_smiling_face:.

Reply:
Thank you! Can you add the link as a comment to the code? :)